examples/website/geospatial/components/parse-file.js

```diff
@@ -0,0 +1,100 @@
+import React, {PureComponent} from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+import {GeoPackageLoader} from '@loaders.gl/geopackage';
+import {FlatGeobufLoader} from '@loaders.gl/flatgeobuf';
+import {load, registerLoaders, selectLoader} from '@loaders.gl/core';
+import {SUPPORTED_FORMATS} from '../examples';
+
+const ErrorFormatHeader = styled.h1`
+  color: red;
+  font-size: 16px;
+`;
+
+registerLoaders([GeoPackageLoader, FlatGeobufLoader]);
+
+const propTypes = {
+  file: PropTypes.object,
+  onFileUploaded: PropTypes.func
+};
+
+const defaultProps = {
+  file: null
+};
+
+export default class ParsedFile extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      fileError: null,
+      data: null
+    };
+  }
+
+  async componentDidMount() {
+    const data = await this.getFileDataUrl();
+    this.setState({data});
+  }
+
+  async getFileDataUrl() {
+    const {file, onFileUploaded} = this.props;
+    const format = this._checkFileFormat(file);
+    if (format) {
+      try {
+        const {arrayBuffer, src} = await this.getLoadedData(file);
```

Comment:
Why do we need all this code? Can't we just call `load()` and let loaders.gl figure if there is not loader?